tests/identityref/run.py

```diff
@@ -1,166 +1,125 @@
-    try:
-      i.test_container.id2 = k[0]
-    except ValueError:
-      passed = False
-    assert passed is k[1], \
-        "id2 leaf was set incorrectly (%s: %s != %s)" % \
-            (k[0], k[1], passed)
-
-  for k in [("grandmother", True), ("mother", True), ("niece", False),
-              ("aunt", True), ("cousin", True), ("daughter", True),
-                ("son", False), ("father", False), ("grandfather", False)]:
-    passed = True
-    try:
-      i.test_container.id3 = k[0]
-    except ValueError:
-      passed = False
-
-    assert passed is k[1], \
-      "id3 leaf was set incorrectly (%s: %s != %s)" % \
-          (k[0], k[1], passed)
-
-  for k in [("daughter", True), ("cousin", False), ("aunt", False)]:
-    passed = True
-    try:
-      i.test_container.id4 = k[0]
-    except ValueError:
-      passed = False
-
-    assert passed is k[1], \
-      "id4 leaf was set incorrectly (%s: %s != %s)" % \
-        (k[0], k[1], passed)
-
-  for k in [("daughter", True), ("cousin", True), ("mother", True),
-            ("aunt", True), ("greatgrandmother", False)]:
-    passed = True
-    try:
-      i.test_container.id5 = k[0]
-    except ValueError:
-      passed = False
-
-    assert passed is k[1], \
-      "id5 leaf was set incorrectly (%s: %S != %s)" % \
-        (k[0], k[1], passed)
-
-  for atype in [("source-dest", True), ("lcaf", True), ("unknown", False)]:
-    passed = True
-    try:
-      i.ak.address_type = atype[0]
-    except ValueError:
-      passed = False
-    assert passed is atype[1], "AK identity inheritance test failed - " + \
-      "%s: %s != %s" % (atype[0], atype[1], passed)
-
-  for k in [("remote:remote-one", True), ("nottherightmodulename:remote-one", False),
-            ("remote:remote-two", True)]:
-    passed = True
-    try:
-      i.test_container.idr1 = k[0]
-    except ValueError:
-      passed = False
-
-    assert passed == k[1], "Did not set an identityref " + \
-      "based on the module-name:value format correctly:" + \
-      "  %s != %s for %s" % (k[1], passed, k[0])
-
-  for tc in [("remote-id", True), ("remote-two:remote-id", True),
-             ("invalid", False)]:
-    passed = True
-    try:
-      i.ietfint.ref = tc[0]
-    except ValueError:
-      passed = False
-
-    assert passed == tc[1], "setting ietfint.ref to %s incorrect " + \
-      "%s != %s" % (tc[0], tc[1], passed)
-
-  if not keepfiles:
-    os.system("/bin/rm %s/bindings.py" % this_dir)
-    os.system("/bin/rm %s/bindings.pyc" % this_dir)
+import unittest
+
+from tests.base import PyangBindTestCase
+
+
+class IdentityRefTests(PyangBindTestCase):
+  yang_files = ['identityref.yang', 'remote-two.yang']
+
+  def setUp(self):
+    self.instance = self.bindings.identityref()
+
+  def test_identityref_leafs_get_created(self):
+    for leaf in ['id1', 'idr1']:
+      with self.subTest(leaf=leaf):
+        self.assertTrue(hasattr(self.instance.test_container, leaf))
+
+  def test_cant_assign_string_to_identityref(self):
```

Comment:
Suggest `test_cant_assign_invalid_string_to_identityref` - since all `identityref` values are strings, it's simply that this one is not a value specified in the YANG file.